Mathlib/RingTheory/MvPolynomial/MonomialOrder.lean

```diff
@@ -71,16 +71,42 @@ variable {R : Type*} [CommSemiring R]
 
 variable (m) in
 /-- the degree of a multivariate polynomial with respect to a monomial ordering -/
-def degree {R : Type*} [CommSemiring R] (f : MvPolynomial σ R) : σ →₀ ℕ :=
+def degree (f : MvPolynomial σ R) : σ →₀ ℕ :=
   m.toSyn.symm (f.support.sup m.toSyn)
 
 variable (m) in
 /-- the leading coefficient of a multivariate polynomial with respect to a monomial ordering -/
-def leadingCoeff {R : Type*} [CommSemiring R] (f : MvPolynomial σ R) : R :=
+def leadingCoeff (f : MvPolynomial σ R) : R :=
   f.coeff (m.degree f)
 
 @[deprecated (since := "2025-01-31")] alias lCoeff := leadingCoeff
 
+theorem leadingCoeff_eq {f : MvPolynomial σ R} : m.leadingCoeff f = f.coeff (m.degree f) := rfl
+
+variable (m) in
+/-- A multivariate polynomial is `Monic` with respect to a monomial order
+if its leading coefficient (for that monomial order) is 1 -/
+def Monic (f : MvPolynomial σ R) : Prop :=
+  m.leadingCoeff f = 1
+
+theorem Monic.def (f : MvPolynomial σ R) : m.Monic f ↔ m.leadingCoeff f = 1 :=
+  Iff.rfl
+
+@[nontriviality] theorem monic_of_subsingleton [Subsingleton R] {f : MvPolynomial σ R} :
+    m.Monic f :=
+  Subsingleton.eq_one (m.leadingCoeff f)
+
+instance Monic.decidable [DecidableEq R] (f : MvPolynomial σ R) :
+    Decidable (m.Monic f) := by
+  unfold Monic; infer_instance
+
+@[simp]
+theorem Monic.leadingCoeff {f : MvPolynomial σ R} (hf : m.Monic f) : m.leadingCoeff f = 1 :=
```

Comment:
```suggestion
theorem Monic.leadingCoeff_eq_one {f : MvPolynomial σ R} (hf : m.Monic f) : m.leadingCoeff f = 1 :=
```
Otherwise I would expect something like `p.Monic -> p.leadingCoeff.Monic` (which could actually typecheck if `p : R[X][X]`

Reply:
The name was chosen following that of [Polynomial.Monic.leadingCoeff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Degree/Definitions.html#Polynomial.Monic.leadingCoeff) !